lib/shopify-cli/commands/connect.rb

```diff
@@ -4,16 +4,30 @@ module ShopifyCli
   module Commands
     class Connect < ShopifyCli::Command
       def call(*)
-        if Project.current
-          @ctx.puts @ctx.message('core.connect.production_warning')
-          org = fetch_org
-          id = org['id']
-          app = get_app(org['apps'])
-          shop = get_shop(org['stores'], id)
-          write_env(app, shop)
-          @ctx.puts(@ctx.message('core.connect.connected', app.first['title']))
-          @ctx.puts(@ctx.message('core.connect.serve', ShopifyCli::TOOL_NAME))
+        project_type = ask_project_type unless Project.has_current?
+
+        @ctx.puts @ctx.message('core.connect.production_warning') if Project.has_current? && Project.current
+
+        env_file = Resources::EnvFile.parse_external_env
+        org = fetch_org
+        id = org['id']
+        app = get_app(org['apps'])
+        shop = get_shop(org['stores'], id)
+
+        write_env(app, shop, env_file[:scopes], env_file[:extra])
+        write_cli_yml(project_type, id) unless Project.has_current?
+
+        @ctx.puts(@ctx.message('core.connect.connected', app.first['title']))
+        @ctx.puts(@ctx.message('core.connect.serve', ShopifyCli::TOOL_NAME))
+      end
+
+      def ask_project_type
+        type_name = CLI::UI::Prompt.ask(@ctx.message("core.connect.project_type_select")) do |handler|
+          ShopifyCli::Commands::Create.all_visible_type.each do |type|
```

Comment:
Since `connect` is a core command right now, I see this for Scripts with this PR:
<img width="696" alt="Screen Shot 2020-06-09 at 6 20 27 AM" src="https://user-images.githubusercontent.com/39379964/84136648-98a3d700-aa19-11ea-8307-775abc1fe7be.png">

I have 2 comments:
1. Why not use it on Production apps?  At least for Scripts, we need to connect to production app in order to deploy scripts to it.
2. The last message "⭑ Run shopify serve to start a local development server" doesn't apply to Scripts project.

Reply:
1. This error is shown when you attempt to connect a project that already has a `.shopify-cli.yml` file. So that project is already a CLI project and doesn't need to be connected. Unless there's another meaning, @tanema? Perhaps we can change the wording to make it clearer.

2. That's fair, I'll remove the start a server stuff, since that wont be consistent for all projects.

Reply:
Ok I'll update this to be something like:
![image](https://user-images.githubusercontent.com/30087610/84932920-021f8780-b0a3-11ea-9ebb-bc9c523d9f6d.png)
Where the second message only shows for Rails apps (wording to be confirmed with @attaboy)

Reply:
Hmm, I'm a little confused by what `connect` actually does, and how it might differ between different project types. To Tim's point, it seems like we may need a different warning message for different project types.

To me, the proposed warning text suggests that the connect command would alter your app's configuration, but that's not really the case, right? What do you mean by "API_KEY details may be updated"?

Reply:
It's potentially updating the fields in the `.env` file (SHOPIFY_API_KEY, SHOPIFY_API_SECRET, SHOP) based on what options you select when it asks you to login to your Partners account and pick an app/shop. So if this app was already live and you chose new/different choices, it could potentially point to a different app in your Partners account than it did before.

Reply:
OK, so based on that description, the proposed text seems misleading to me.

The problem as I understand it is: if you connect to a production app, and the current project has an app URL set locally which differs from the app's, running **`serve`** could break the production app.

And this specific problem only occurs for node and rails app projects, right?

I like Tim's idea: for those project types, we add a warning, and I think the warning should say:

`{{!}} Warning: if you have connected to an {{bold:app in production}}, running {{command:serve}} may update the app URL and cause an outage.`